server/src/models/User.js

```diff
@@ -26,21 +26,22 @@ class User extends uniqueFunc(Model) {
   static get jsonSchema() {
     return {
       type: "object",
-      required: ["email"],
+      required: ["email", "username"],
 
       properties: {
         email: { type: "string", pattern: "^\\S+@\\S+\\.\\S+$" },
+        username: { type: "string", minLength: 3, maxLength: 25 },
         cryptedPassword: { type: "string" },
       },
     };
   }
 
-  $beforeInsert() {
-    return this.$checkUniqueness("email");
+  $beforeInsert = () => {
+    return this.$checkUniqueness("email")
   }
 
-  $beforeUpdate() {
-    return this.$checkUniqueness("email");
+  $beforeUpdate = () => {
+    return this.$checkUniqueness("email")
   }
```

Comment:
not sure I understand the addition of this. Objection-unique should be checking this already